app/models/synonym.rb

```diff
@@ -30,38 +30,70 @@
+  require "arel-helpers"
+
+  include ArelHelpers::ArelTable
+
   has_many :names
 
-  # Mightly cronjob to ensure that no synonym records accidentally got deleted.
+  # Nightly cronjob to ensure that no synonym records accidentally got deleted.
   # This actually happened to Fungi itself(!)  Not sure how it happened, and
   # obviously I'd prefer to fix the cause.  But meanwhile, might as well keep
   # the site working...
   def self.make_sure_all_referenced_synonyms_exist
     msgs = []
-    references = Name.connection.select_values(%(
-      SELECT DISTINCT synonym_id FROM names
-      WHERE synonym_id IS NOT NULL
-      ORDER BY synonym_id ASC
-    ))
-    records = Name.connection.select_values(%(
-      SELECT id FROM synonyms ORDER BY id ASC
-    ))
+    # reference_select = arel_select_referenced_synonyms
+    # # puts(reference_select.to_sql)
+    # references = Name.connection.select_values(reference_select.to_sql)
+    references = Name.select(:synonym_id).where.not(synonym: nil).distinct.
+                 order(:synonym_id).pluck(:synonym_id)
+    # puts(references.join(",").to_s)
+
+    # SELECT id FROM synonyms ORDER BY id ASC
+    # record_select = Synonym.select(:id).order(Synonym[:id].asc)
+    # puts(record_select.to_sql)
+    # records = Name.connection.select_values(record_select.to_sql)
+    records = Synonym.all.order(id: :asc).pluck(:id)
+    # puts(records.join(",").to_s)
     unused  = records - references
     missing = references - records
+    # puts(unused.join(",").to_s)
+    # puts(missing.join(",").to_s)
+
     if unused.any?
-      Name.connection.execute(%(
-        DELETE FROM synonyms
-        WHERE id IN (#{unused.map(&:to_s).join(",")})
-      ))
+      Synonym.where(id: unused).delete_all
+
       msgs << "Deleting #{unused.count} unused synonyms: #{unused.inspect}"
     end
     if missing.any?
-      Name.connection.execute(%(
-        INSERT INTO synonyms (id) VALUES
-        #{missing.map { |id| "(#{id})" }.join(",")}
-      ))
+      insert_manager = arel_insert_missing_synonyms(missing)
+      # puts(insert_manager.to_sql)
+      Name.connection.execute(insert_manager.to_sql)
       msgs << "Restoring #{missing.count} missing synonyms: #{missing.inspect}"
     end
     msgs
   end
+
+  # SELECT DISTINCT synonym_id FROM names
+  # WHERE synonym_id IS NOT NULL
+  # ORDER BY synonym_id ASC
+  # private_class_method def self.arel_select_referenced_synonyms
+  #   names = Name.arel_table
+
+  #   Name.select(:synonym_id).distinct.
+  #     where(names[:synonym_id].not_eq(nil)).
+  #     order(names[:synonym_id].asc)
+  # end
+
+  # INSERT INTO synonyms (id) VALUES
+  # #{missing.map { |id| "(#{id})" }.join(",")}
+  private_class_method def self.arel_insert_missing_synonyms(missing)
+    syn = Synonym.arel_table
+
+    Arel::InsertManager.new.tap do |manager|
+      manager.into(syn)
+      manager.columns << syn[:id]
+      manager.values = manager.create_values(missing, syn[:id])
+    end
+  end
```

Comment:
Modify this and move it to app/models/abstract_model.rb as follows.
Moving it and un-privatizing it may seem premature, but surely it can be used as-is in other models to convert sql `INSERT INTO` to arel. (You're probably ahead of me on this.)
```ruby
  ##############################################################################
  #
  #  :section: arel
  #
  ##############################################################################

  # temporary, partial substitute for Rails 6 insert_all
  # Inserts multiple records into the db in a single SQL INSERT.
  # Does not instantiate any models nor trigger AR callbacks or validations.
  # Passed values go through AR type casting and serialization.
  # Usage example:
  #   records = Synonym.all.order(id: :asc).pluck(:id)
  #   unused  = records - references
  #   missing = references - records
  #   Name.connection.execute(arel_insert_all(missing).to_sql)
  def self.arel_insert_all(values)
    table = arel_table

    Arel::InsertManager.new.tap do |manager|
      manager.into(table)
      manager.columns << table[:id]
      manager.values = manager.create_values(values, table[:id])
    end
  end
```